src/token/api/src/types.rs

```diff
@@ -163,3 +167,22 @@ pub struct PaginatedResult {
 
 pub type TxId = u64;
 pub type Cycles = u64;
+
+/// `TokenHolder` is the holder of a token. This is used to identify the holder of a token in the form of `AccountIdentifier`.
+pub type TokenHolder = AccountIdentifier;
+
+/// `TokenReceiver` is the recipient of a token. This is used to identify the recipient of a token in the form of `AccountIdentifier`.
+pub type TokenReceiver = AccountIdentifier;
```

Comment:
In this case `TokenHolder` and `TokenReceiver` is the same type, which is not ideal. If you want to have the advantage of type checking, its better to use `struct TokenHolder(AccountIdentifier)`, `struct TokenReceiver(AccountIdentifier)` instead. Otherwise using just `AccountIdentifier` is actually better, because it makes the possible place of an error more explicit.

Reply:
IMO `TokenHolder` for the `from` is easier than being `Account Identifier`. It will be easier to follow and understand than using `AccountIdentifier` in both `to` and `from` fields, am gonna take a look in making a struct of the advantages of type-checking.